src/controlCenter/widgets/baseWidget.vala

```diff
@@ -93,15 +103,20 @@ namespace SwayNotificationCenter.Widgets {
             for (int i = 0; i < actions.get_length (); i++) {
                 string label = actions.get_object_element (i).get_string_member_with_default ("label", "label");
                 string command = actions.get_object_element (i).get_string_member_with_default ("command", "");
+                string t = actions.get_object_element (i).get_string_member_with_default ("type", "normal");
+                ButtonType type = ButtonType.parse (t);
+                bool active = actions.get_object_element (i).get_boolean_member_with_default ("active", false);
                 res[i] = Action () {
                     label = label,
-                    command = command
+                    command = command,
+                    type = type,
+                    active = active
                 };
             }
             return res;
         }
 
-        protected void execute_command (string cmd) {
+        public static void execute_command (string cmd) {
```

Comment:
We should instead move this into `functions.vala` if it's going to be used outside of the base class.

While doing so, we should also move to the way that the `Script` class executes commands but using `Process.spawn_command_line_async` instead. Then we can also add a list of strings as a environment variable list parameter to fix the script variable issue posted below :)